src/components/refactor/dialogs/line-creation/line-limits-pane/line-limits-pane-utils.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import {
+    LIMITS,
+    CURRENT_LIMITS_1,
+    CURRENT_LIMITS_2,
+    PERMANENT_LIMIT,
+    TEMPORARY_LIMITS,
+    TEMPORARY_LIMIT_NAME,
+    TEMPORARY_LIMIT_DURATION,
+    TEMPORARY_LIMIT_VALUE,
+} from 'components/refactor/utils/field-constants';
+import yup from '../../../utils/yup-config';
+import { areArrayElementsUnique } from '../../../utils/utils';
+import { sanitizeString } from '../../../../dialogs/dialogUtils';
+
+const temporaryLimitsTableValidationSchema = () => ({
+    [PERMANENT_LIMIT]: yup
+        .number()
+        .nullable()
+        .positive('permanentCurrentLimitGreaterThanZero'),
+    [TEMPORARY_LIMITS]: yup
+        .array()
+        .of(
+            yup.object().shape({
+                [TEMPORARY_LIMIT_NAME]: yup.string().required(),
+                [TEMPORARY_LIMIT_DURATION]: yup.number().nullable().min(0),
+                [TEMPORARY_LIMIT_VALUE]: yup.number().nullable().positive(),
+            })
+        )
+        .test('distinctNames', 'TemporaryLimitNameUnicityError', (array) => {
+            const namesArray = array
+                .filter((l) => !!l[TEMPORARY_LIMIT_NAME])
```

Comment:
we check name unicity only for non-empty names: that way, empty name fields are red (in error) instead of having an error in the bottom "Temporary limit names must be unique in the table" (when we have more than 1 empty name)